common/lib/xmodule/xmodule/js/spec/capa/display_spec.coffee

```diff
@@ -424,6 +429,47 @@ describe 'Problem', ->
       @problem.reset()
       expect($('.notification-gentle-alert .notification-message').text()).toEqual("Error on reset.")
 
+    it 'tests that reset does not enable the submit button when submit is originally disabled', (done) ->
+      deferred = $.Deferred()
+      self = this
+      curr_html = @problem.el.html()
+      runs = ->
+        spyOn($, 'postWithPrefix').and.callFake (url, answers, callback) ->
+          promise = undefined
+          callback(success: 'correct', html: curr_html)
+          promise = always: (callable) ->
+            callable()
+        expect(@problem.submitButton).toHaveAttr('disabled')
+        @problem.reset()
+        if jQuery.active == 0
+          deferred.resolve()
+        deferred.promise()
+
+      runs.call(self).then(->
```

Comment:
I'm not convinced this test point is working correctly. Since the spyOn is defined in the "runs", and you call reset before you ever call "runs", how does the spy actually do anything? Shouldn't the spyOn be handing the ajax call for runs?

I think this could all be simplified somewhat. Take a look at the specific "reset" test points above.